src/Telemetry/AzureDevOpsTimeline/Program.cs

```diff
@@ -52,14 +54,20 @@ private static void Main()
                                     parallelRequests = 5;
                                 }
 
-                                return new AzureDevOpsClient(
-                                    baseUrl: c["AzureDevOpsUrl"],
-                                    organization: c["AzureDevOpsOrganization"],
-                                    maxParallelRequests: parallelRequests,
-                                    accessToken: c["AzureDevOpsAccessToken"]
-                                );
+                                return new AzureDevOpsClientOptions
+                                {
+                                    BaseUrl = c["AzureDevOpsUrl"],
+                                    Organization = c["AzureDevOpsOrganization"],
+                                    MaxParallelRequests = parallelRequests,
+                                    AccessToken = c["AzureDevOpsAccessToken"]
+                                };
                             });
 
+                            services.AddSingleton<ISystemClock, SystemClock>();
+                            services.AddSingleton<RetryAfterHandler>();
```

Comment:
It's weird to inject a single, named handler type.  It makes more sense to inject it as an "AzureDevOpsDelegatingHandler", and then take an IEnumerable of those, and hook them all up.  That way you can choose which handlers to include without having to change all the consumers.

The AzureDevOpsClient doesn't care about this _specific_ handler, it just wants to be told how to make HTTP requests.

Reply:
I injected it as a DelegatingHandler, do you think that is too wide?

Reply:
That's a _bit_ wide, but... shouldn't be too bad? I can't see a reason anyone else would inject a "DelegatingHandler" other than for it to be used for all HTTP communication.

Reply:
It might be cleaner to use HttpClientFactory to construct the HttpClients with whatever delegate handlers are needed.

Reply:
I created `AzureDevOpsHttpClientFactory` which collects all `AzureDevOpsDelegatingHandlers` and creates a `HttpClient` with them. Hope I understood what you meant